flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -293,7 +295,15 @@ public static void main(String[] args) throws Exception {
 			LOG.info("Cannot determine the maximum number of open file descriptors");
 		}
 
-		runTaskManagerSecurely(args, ResourceID.generate());
+		final Configuration config = loadConfiguration(args);
+		final String resourceIDPrefix = config.get(TaskManagerOptions.TASK_MANAGER_RESOURCE_ID_PREFIX);
+		ResourceID taskManagerResourceID;
```

Comment:
Could we wrap the logic to get the `resourceID` into a separate method? Also we could add some tests for it.
* Do not specify prefix
* Specify prefix
  * get more than once and should have two different resource id